fiftyone/utils/eval/classification.py

```diff
@@ -0,0 +1,499 @@
+
+    ytrue, ypred, confs = samples.aggregate(
+        [foa.Values(gt), foa.Values(pred), foa.Values(pred_conf)]
+    )
+
+    if eval_field:
+        samples._add_field_if_necessary(eval_field, fof.BooleanField)
+        samples.set_field(eval_field, F(gt) == F(pred)).save(eval_field)
+
+    # Equivalent with loops
+    # ytrue = []
+    # ypred = []
+    # confs = []
+    # with fou.ProgressBar() as pb:
+    #     for sample in pb(samples.select_fields([pred_field, gt_field])):
+    #         gt_label = sample[gt_field].label
+    #         pred_label = sample[pred_field].label
+    #         pred_conf = sample[pred_field].confidence
+    #         ytrue.append(gt_label)
+    #         ypred.append(pred_label)
+    #         confs.append(pred_conf)
+    #         if eval_field:
+    #             sample[eval_field] = gt_label == pred_label
+    #             sample.save()
+
+    return ClassificationResults(
+        ytrue, ypred, confs, classes=classes, missing=missing
+    )
+
+
+def evaluate_binary_classifications(
+    samples, classes, pred_field, gt_field="ground_truth", eval_field=None,
+):
+    """Evaluates the binary classification predictions in the given samples
+    with respect to the specified ground truth labels.
+
+    Any missing ground truth or prediction labels are assumed to be examples of
+    the negative class (with zero confidence, for predictions).
+
+    If an ``eval_field`` is specified, this method will record the TP/FP/FN/TN
+    status of each prediction in this field.
+
+    Args:
+        samples: a :class:`fiftyone.core.collections.SampleCollection`
+        classes: the ``(neg_label, pos_label)`` label strings for the task
+        pred_field: the name of the field containing the predicted
+            :class:`fiftyone.core.labels.Classification` instances to evaluate
+        gt_field ("ground_truth"): the name of the field containing the ground
+            truth :class:`fiftyone.core.labels.Classification` instances
+        eval_field (None): the name of a field in which to record whether each
+            prediction is correct
+
+    Returns:
+        a :class:`BinaryClassificationResults`
+    """
+    pos_label = classes[-1]
+
+    gt = gt_field + ".label"
+    pred = pred_field + ".label"
+    pred_conf = pred_field + ".confidence"
+
+    ytrue, ypred, confs = samples.aggregate(
+        [foa.Values(gt), foa.Values(pred), foa.Values(pred_conf)]
+    )
+
+    if eval_field:
+        samples._add_field_if_necessary(eval_field, fof.StringField)
+        samples.set_field(
+            eval_field,
+            F().switch(
+                {
+                    (F(gt) == pos_label) & (F(pred) == pos_label): "TP",
+                    (F(gt) == pos_label) & (F(pred) != pos_label): "FN",
+                    (F(gt) != pos_label) & (F(pred) != pos_label): "TN",
+                    (F(gt) != pos_label) & (F(pred) == pos_label): "FP",
+                }
+            ),
+        ).save(eval_field)
```

Comment:
Here's a fairly complex update that *can* be performed purely as an aggregation pipeline

Reply:
:rocket: